src/Request/Parameters/Parameter.php

```diff
@@ -73,6 +74,18 @@ public function required(): self
         return $this;
     }
 
+    /**
+     * Marks the value of the parameter as required + non-empty and
+     * throws an exception on validate if the value is missing.
+     */
+    public function requiredNonEmpty(): self
+    {
+        if (empty($this->value)) {
```

Comment:
Does this handle null value?

Reply:
yes it does!

Reply:
Have we added the test case for this i.e. null param value?

Reply:
yes its covered with both `null` and `""` values